packages/core/src/base/staticClient.ts

```diff
@@ -93,6 +93,30 @@ export const logResponse = (response: Response): void => {
   );
 };
 
+/**
+ * Exports a minipass-fetch Headers object as a map.
+ * 
+ * minipass-fetch Headers class has a function to convert a Headers object into
+ * a map. But that function creates an array of values for each header. 
+ * `http-cache-semantics`, a package used by `make-fetch-happen` to manipulate
+ * headers expects them to be comma separated strings.
+ *
+ * @param headers - a minipass-fetch Headers object
+ *
+ * @returns a map of headers
+ */
+export function exportHeadersAsMap(headers: Headers) {
+  const res = {};
+  const keysIterator: HeadersIterator = headers.keys();
+  let key = keysIterator.next();
+  while (!key.done) {
```

Comment:
This works, but why not something like

```
export function exportHeadersAsMap(headers: Headers) {
  const res = {};
  const headersIterator: HeadersIterator = headers.entries();
  for (const [key, value] of headersIterator) {
    res[key] = value;
  }
  return res;
}
```
?

Just think it is cleaner to read

Reply:
The cleanest solution would actually be to just use `Object.fromEntries(headers)`, but that's not available in node 10. Lodash does provide a convenient alternative, though: `_.fromPairs([...headers])`.

Generally, I agree that we should be trying to avoid lodash whenever possible, but I think in this case it does provide a clean solution that allows us to avoid writing our own custom function.

Reply:
```
const res = {};
headers.forEach((val, key, obj) => res[key] = obj.get(key));
return res;
```

Reply:
> The cleanest solution would actually be to just use `Object.fromEntries(headers)`, but that's not available in node 10. Lodash does provide a convenient alternative, though: `_.fromPairs([...headers])`.
Those give you the same result as just calling headers.raw(). The problem this addresses is calling 'get' which does a join on the value before returning.

Reply:
`_.fromPairs()` works. `.raw()` only creates arrays of values whereas `...headers` creates arrays of `key + value` which is what `fromPairs` needs to create an object of key value strings.